code/game/objects/items/gift.dm

```diff
@@ -8,22 +8,31 @@
  * Gifts
  */
 
-GLOBAL_LIST_EMPTY(possible_gifts)
+GLOBAL_LIST_EMPTY(possible_toy_gifts)
+GLOBAL_LIST_EMPTY(possible_anything_gifts)
 
 /obj/item/a_gift
-	name = "gift"
-	desc = "PRESENTS!!!! eek!"
+	name = "christmas gift"
+	desc = "It could be anything!"
 	icon = 'icons/obj/storage.dmi'
 	icon_state = "giftdeliverypackage3"
 	item_state = "gift"
 	resistance_flags = FLAMMABLE
+	var/obj/item/gift_type
 
-/obj/item/a_gift/New()
-	..()
+/obj/item/a_gift/Initialize(mapload)
+	. = ..()
 	pixel_x = rand(-10,10)
 	pixel_y = rand(-10,10)
 	icon_state = "giftdeliverypackage[rand(1,5)]"
 
+	gift_type = get_gift_type()
+
+/obj/item/a_gift/examine(mob/user)
+	. = ..()
+	if(isobserver(user) || (user && user.mind && user.mind.special_role == "Santa"))
```

Comment:
Please don't add more special role use cases. Check for suit/spell/magic santa eyes/make new antag but not special role please.

Reply:
I'm not adding the case, it was already used to stop him opening his own gifts.

...santa eyes?

Reply:
Oh, if making a new antagonist is easy, I will do that.

Reply:
For santa it would basically be just equip outfit and give objective. You can throw in tracking presents given if you're fancy.